test/clj/rems/test_browser.clj

```diff
@@ -315,83 +424,170 @@
-
-    (testing "changed language must have been saved for user"
-      (logout)
-      (change-language :en)
-      (wait-visible *driver* {:tag :h1 :fn/text "Welcome to REMS"})
-      (delete-cookies *driver*)
-      (login-as "alice")
-      (wait-visible *driver* {:tag :h1, :fn/text "Aineistoluettelo"}))
-
-    (testing "changing language while logged in"
-      (change-language :en)
-      (wait-visible *driver* {:tag :h1 :fn/text "Catalogue"}))
-    (is true))) ; avoid no assertions warning
+  (let [driver (get-driver)]
+    (et/with-postmortem driver {:dir reporting-dir}
+      (testing "default language is English"
+        (doto driver
+          (et/go (get-url))
+          (et/wait-visible {:tag :h1 :fn/text "Welcome to REMS"})
+          (login-as "alice")
+          (et/wait-visible {:tag :h1 :fn/text "Catalogue"})
+          (wait-page-loaded)))
+
+      (testing "changing language while logged out"
+        (doto driver
+          (logout)
+          (et/wait-visible {:tag :h1 :fn/text "Welcome to REMS"})
+          (change-language :fi)
+          (et/wait-visible {:tag :h1 :fn/text "Tervetuloa REMSiin"})))
+
+      (testing "changed language must persist after login"
+        (doto driver
+          (login-as  "alice")
+          (et/wait-visible {:tag :h1 :fn/text "Aineistoluettelo"})
+          (wait-page-loaded)))
+
+      (testing "wait for language change to show in the db"
+        (wait-predicate driver #(= :fi (:language (user-settings/get-user-settings "alice")))))
+
+      (testing "changed language must have been saved for user"
+        (doto driver
+          (logout)
+          (change-language :en)
+          (et/wait-visible {:tag :h1 :fn/text "Welcome to REMS"})
+          (et/delete-cookies)
+          (login-as "alice")
+          (et/wait-visible {:tag :h1 :fn/text "Aineistoluettelo"})))
+
+      (testing "changing language while logged i"
+        (change-language driver :en)
+        (et/wait-visible driver {:tag :h1 :fn/text "Catalogue"}))
+      (is true)))) ; avoid no assertions warning
+
+;; TODO: driver is passed to scroll-and-click, maybe create higher level overload with shorter name too
+
+(defn slurp-fields [driver selector]
+  (->> (for [row (et/query-all driver [selector {:fn/has-class :row}])
+             :let [k (et/get-element-text-el driver (et/child driver row {:tag :label}))
+                   v (et/get-element-text-el driver (et/child driver row {:css ".form-control"}))]]
+         [k (str/trim v)])
+       (into {})))
+
+
+(defn test-create-license []
+  (let [driver (get-driver)]
+    (et/with-postmortem driver {:dir reporting-dir}
+      (doto driver
+        (go-to-admin-licenses)
+        (scroll-and-click  :create-license)
+        (et/wait-visible  {:tag :h1 :fn/text "Create license"})
+        (select-option  "Organization" "nbn")
+        (scroll-and-click :licensetype-link)
+        (fill-form-field "License name" (str (:license-name @test-context) " EN") {:index 1})
```

Comment:
do you just want to demo the `:index` feature here or is there a reason you can't `(fill-form-field "License name (EN)" ...)`?

Reply:
It's always called "License name" since the language is in the header before this spot. This is because the type selection affects the content of each place. A possible refactoring of the screen is to not use the header but embed the language in each label instead. Then there is no need for the index. I'll not do this yet.